### ISSUE TITLE
Checkbox, ContextMenu and Child node deletion

### ISSUE THREAD

**Issue by webmutation** [vakata/jstree#724]
Hi, 

I think there is a bug when you use the check-box and context menu together. When  you right click for the context menu you get all items selected, so if I delete one of the child nodes I actually end up deleting the whole tree... so check-box does not appear to make a distinction between left and right click. Is this by design or is it a bug?

![tree](https://cloud.githubusercontent.com/assets/761367/3325154/01823f84-f78b-11e3-9f7c-c3bdc1399fb4.png)

So in summary to test this you right click a child node and create a new node, you then create a child node of that child and all the above nodes get selected, however I only wanted to delete that newly created child node. Not the whole branch.
